bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/ValueFactory.java

```diff
@@ -72,10 +76,12 @@ public static Value createValueState(ChannelConfig config, String channelTypeID)
                 value = new ColorValue(colorMode, config.on, config.off, config.onBrightness);
                 break;
             case MqttBindingConstants.SWITCH:
-                value = new OnOffValue(config.on, config.off);
+                value = new OnOffValue(config.onState == null && on != null ? List.of(on) : config.onState,
+                        config.offState == null && off != null ? List.of(off) : config.offState);
                 break;
             case MqttBindingConstants.CONTACT:
-                value = new OpenCloseValue(config.on, config.off);
+                value = new OpenCloseValue(config.onState == null && on != null ? List.of(on) : config.onState,
+                        config.offState == null && off != null ? List.of(off) : config.offState);
                 break;
             case MqttBindingConstants.ROLLERSHUTTER:
                 value = new RollershutterValue(config.on, config.off, config.stop, config.onState, config.offState,
```

Comment:
do you not also need to handle multiple values for Rollershutter here?

Reply:
This is implemented inside `RollershutterValue`: It gets the `List<String>` onState and offState passed.